R/zchunk_LA122.gasproc_refining.R

```diff
@@ -42,100 +42,370 @@ module_energy_LA122.gasproc_refining_DISABLED <- function(command, ...) {
+      select(-sector) %>%
+      rename(sector = biodiesel) %>%
+      bind_rows(BIOMASS_LIQUIDS_Ethanol) -> L122.out_EJ_R_biofuel_Yh
+
+    # Adding fuels to appropiate sectors and regions in L122.out_EJ_R_biofuel_Yh from calibrated_techs
+    calibrated_techs %>%
+      select(sector, fuel) %>%
+      distinct(sector, .keep_all = TRUE) %>%
+      right_join(L122.out_EJ_R_biofuel_Yh, by = "sector") %>%
+      select(GCAM_region_ID, sector, fuel = fuel.x, year, value) -> L122.out_EJ_R_biofuel_Yh
+
+
+    # Inputs to biofuel production are region-specific
+    # Because some have multiple inputs, repeat coefficient table by number of regions and then subset only the applicable combinations
+    L122.globaltech_coef %>%
+      filter(sector %in% L122.out_EJ_R_biofuel_Yh$sector) %>%
+      repeat_add_columns(tibble::tibble(GCAM_region_ID = A_regions$GCAM_region_ID)) -> L122.biofuel_coef_repR
+
+    # subset L122.biofuel_coef_repR based on A_regions by region and sector (ethanol)
+    L122.biofuel_coef_repR %>%
+      semi_join(select(A_regions, GCAM_region_ID, sector = ethanol), by = c("sector", "GCAM_region_ID")) -> L122.biofuel_coef_repR_Ethanol
+
+    # subset L122.biofuel_coef_repR based on A_regions by region and sector (biodisiel). Then adding ethanol sector (L122.biofuel_coef_repR_Ethanol)
+    L122.biofuel_coef_repR %>%
+      semi_join(select(A_regions, GCAM_region_ID, sector = biodiesel), by = c("sector", "GCAM_region_ID")) %>%
+      bind_rows(L122.biofuel_coef_repR_Ethanol) -> L122.biofuel_coef_R
+
+    # Build table of inputs to biofuel production (IO coefs times output)
+    L122.biofuel_coef_R %>%
+      rename(fuelInput = fuel, valueInput = value) %>%
+      left_join(L122.out_EJ_R_biofuel_Yh, by = c("GCAM_region_ID", "sector","year")) %>%
+      mutate(value_out  = valueInput*value) %>%
+      select(-value, -valueInput, -fuel) %>%
+      rename(value = value_out, fuel = fuelInput) %>%
+      select(GCAM_region_ID, sector, fuel, year, value) -> L122.in_EJ_R_biofuel_F_Yh
+
+    # GAS AND COAL TO LIQUIDS
+    # Create L122.out_EJ_R_gtlctl_Yh from L1011.en_bal_EJ_R_Si_Fi_Yh for gas to liquids (gtl) and coal to liquids (ctl) sectors
+    L1011.en_bal_EJ_R_Si_Fi_Yh %>%
+      filter(sector == "out_gtl" | sector == "out_ctl") %>%
+      mutate(sector = if_else(sector=="out_gtl", "gtl", "ctl")) -> L122.out_EJ_R_gtlctl_Yh
+
+    # GTL and CTL inputs (L122.in_EJ_R_gtlctl_F_Yh): derived as output times exogenous input-output coefficients
+    # Interpolate gas processing IO coefs to all historical years and match in the fuel name
+    L122.globaltech_coef %>%
+      filter(sector == "gtl"| sector == "ctl") -> L122.gtlctl_coef
+
+    L122.out_EJ_R_gtlctl_Yh %>%
+      rename(valueInput = value) %>%
+      left_join(L122.gtlctl_coef, by = c("sector", "fuel", "year")) %>%
+      mutate(value_out = valueInput * value) %>%
+      select(-value, -valueInput) %>%
+      rename(value = value_out) %>%
```

Comment:
Could simplify above three lines to one I think: `mutate(value = valueInput * value)` (since `valueInput` will be dropped in line 144 below)